src/main/starlark/builtins_bzl/common/cc/cc_import.bzl

```diff
@@ -153,12 +153,18 @@ def _cc_import_impl(ctx):
             linker_inputs = depset([linker_input]),
         )
 
+    additional_make_variable_substitutions = cc_helper.get_toolchain_global_make_variables(cc_toolchain)
+    additional_make_variable_substitutions.update(cc_helper.get_cc_flags_make_variable(ctx, feature_configuration, cc_toolchain))
+
     (compilation_context, _) = cc_common.compile(
         actions = ctx.actions,
         feature_configuration = feature_configuration,
         cc_toolchain = cc_toolchain,
         public_hdrs = ctx.files.hdrs,
-        includes = ctx.attr.includes,
+        system_includes = cc_helper.system_include_dirs(
```

Comment:
Why does this change need to be so complex? Where is `ctx.attr.includes` even read?

Please add a test for it.

This is a breaking change, right?

Reply:
> Why does this change need to be so complex?

"complex" is a subjective expression, so I find it difficult to answer. If you could tell me a simpler approach, I would appreciate it.

> Where is `ctx.attr.includes` even read?

It's referred to inside `cc_helper.system_include_dirs`.

> Please add a test for it.

Sure, but where/how am I supposed to add a test? Indeed I know there's a directory `src/main/starlark/tests/builtins_bzl/cc/` which looks corresponding to `src/main/starlark/builtins_bzl/common/cc`, but there's no test but that of `cc_shared_library` (but I may be looking at wrong directory).

> This is a breaking change, right?

It's supposed to change the behavior of the public attribute `cc_import.includes`. Should I avoid changing this attribute, by adding another attribute like `system_includes` instead?

Reply:
By complex I mean, why not
`system_includes = ctx.attr.includes`?

There seem to be 2 things going on. The new code also processes build variables, whereas previous didn't.

Incompatible changes need extra effort. I need to test internal repository for breakages and if there are any, fix them.

Also Bazel community will need to fix incompatibilities.

As such I don't think the motivation (inconsistence with cc_library) is high enough to justify the effort.

Reply:
This change, complicated or not, actually brings the code in-line with the documentation.  

https://bazel.build/reference/be/c-cpp#cc_import_args
> includes	
> List of strings; default is []
> 
> List of include dirs to be added to the compile line. Subject to ["Make variable"](https://bazel.build/reference/be/make-variables) substitution. Each string is prepended with the package path and passed to the C++ toolchain for expansion via the "include_paths" CROSSTOOL feature. A toolchain running on a POSIX system with typical feature definitions will produce -isystem path_to_package/include_entry. This should only be used for third-party libraries that do not conform to the Google style of writing #include statements. Unlike [COPTS](https://bazel.build/reference/be/c-cpp#cc_binary.copts), these flags are added for this rule and every rule that depends on it. (Note: not the rules it depends upon!) Be very careful, since this may have far-reaching effects. When in doubt, add "-I" flags to [COPTS](https://bazel.build/reference/be/c-cpp#cc_binary.copts) instead.
> The default include path doesn't include generated files. If you need to #include a generated header file, list it in the srcs.

Please either fix the functionality, (which is probably what most everyone in the bazel community would want), or fix the documentation.